bindings/node/lib/clientEncryption.js

```diff
@@ -42,7 +42,34 @@ module.exports = function(modules) {
      */
     createDataKey(provider, options, callback) {
       if (typeof options === 'function') (callback = options), (options = {});
-      options = options || {};
+      options = Object.assign({}, options);
+
+      // To avoid using libbson inside the bindings, we pre-serialize
+      // any keyAltNames here.
+      if (options.keyAltNames) {
```

Comment:
nit: does it make sense to move this all out into its own function to keep readability simple here?

Reply:
I think I can refactor this into a helper.